### PR TITLE
[CoreBundle] Honeypot form basic implementation.

### PR DESCRIPTION
**Pull request by ngodfraind**
This is a (very) basic implementation of a honeypot I'd like to try (and combine it with the captcha to reduce the amount of bots going through).

It can be improved later. I plan to use it on claroline.net and see how it goes.

My prefered method for catching bot works with javascript (it's simply adding a parameter in the querystring when the user clicks on a submit button... this parameter is then checked and if it doesn't exists, we assume it's a bot) but it's not very generic and I don't think it'll be usable on a large scale.

it's related to #388 but honeypots are less of an accessibility problem.

### CODE REVIEW

File: main/core/Form/Field/HoneypotType.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            'required' => false,
+            'mapped' => false,
+            'data' => '',
+            'label' => ' ',
+            'attr' => array(
+                'autocomplete' => 'off',
+                'tabindex' => -1,
+                // Fake `display:none` css behaviour to hide input
+                // as some bots may also check inputs visibility
+                'style' => 'position: fixed; left: -100%; top: -100%;',
```
Comment (stefk) [inline main/core/Form/Field/HoneypotType.php]: Accessiblity issue. Screen readers users should be warned somehow that the field must be left empt, eg by using an appropriate labe/aria-label (probably without giving a too obvious hint to bots).

**Comment by stefk** [comment]
What's the related issue? I can't be #366.

**Comment by ngodfraind** [comment]
it's #388 sorry

**Comment by ptsavdar** [comment]
Besides accessibility issue mentioned by @stefk I'd say LGTM

**Comment by LaurentGruber** [comment]
LGTM

**Comment by LaurentGruber** [comment]
LGTM

**Pull request merged**